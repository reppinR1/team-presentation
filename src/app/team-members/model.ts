export interface TeamMembersResponse {
  data: TeamMembersResponseData[];
}

export interface TeamMembersResponseData {
  id: string;
  attributes: Attributes;
  type: string;
}

export interface TeamMember {
  imageUrl: ImageUrl;
  block: Block;
}

export interface MemberCards {
  first: MemberCard;
  second: MemberCard;
  third: MemberCard;
}

interface Attributes {
  title: string;
  memberCards: MemberCards;
}

interface MemberCard {
  imageUrl: ImageUrl;
  block: Block;
}

interface ImageUrl {
  w200: string;
  w400: string;
  w1080: string;
  w1920: string;
  w2560: string;
}

interface Block {
  title: string;
  description: string;
  link: string;
  text: string;
}
