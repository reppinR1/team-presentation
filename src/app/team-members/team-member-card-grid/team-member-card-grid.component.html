<ng-container *ngIf="memberCards; else loading">
  <section class="fluid-grid" data-test="card-grid">
    <app-team-member-card
      [memberCard]="memberCard.value"
      *ngFor="let memberCard of memberCards | keyvalue"
    >
    </app-team-member-card>
  </section>
</ng-container>
<ng-template #loading>
  <section data-test="loading-section">Loading...</section></ng-template
>
