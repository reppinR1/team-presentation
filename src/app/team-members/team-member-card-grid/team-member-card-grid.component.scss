:host {
  display: block;
  width: 100%;
}

.fluid-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(240px, 1fr));
  grid-gap: 8px;
  justify-items: center;
}
