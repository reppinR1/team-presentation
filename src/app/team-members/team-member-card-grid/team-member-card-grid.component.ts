import { Component, Input } from '@angular/core';
import { MemberCards } from '../model';

@Component({
  selector: 'app-team-member-card-grid',
  templateUrl: './team-member-card-grid.component.html',
  styleUrls: ['./team-member-card-grid.component.scss'],
})
export class TeamMemberCardGridComponent {
  @Input() memberCards: MemberCards | null;
}
