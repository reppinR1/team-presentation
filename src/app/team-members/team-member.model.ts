export interface TeamMembersResponse {
  data: TeamMembersResponseData[];
}

export interface TeamMembersResponseData {
  id: string;
  attributes: Attributes;
  type: string;
}

export interface MemberCards {
  first: MemberCard;
  second: MemberCard;
  third: MemberCard;
}

export interface MemberCard {
  imageUrl: ImageUrl;
  block: Block;
}

export interface ImageUrl {
  w200: string;
  w400: string;
  w1080: string;
  w1920: string;
  w2560: string;
}

interface Attributes {
  title: string;
  memberCards: MemberCards;
}

interface Block {
  title: string;
  description: string;
  link: string;
  text: string;
}
