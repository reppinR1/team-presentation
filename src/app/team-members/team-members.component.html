<app-team-member-card-grid [memberCards]="memberCards$ | async">
</app-team-member-card-grid>
h
