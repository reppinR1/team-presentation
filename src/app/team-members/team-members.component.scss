:host {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.title {
  color: var(--text-secondary);
  font-weight: bold;
}
