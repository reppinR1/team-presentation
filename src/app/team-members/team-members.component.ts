import { ChangeDetectionStrategy, Component } from '@angular/core';
import { Observable, of } from 'rxjs';
import { MemberCards } from './team-member.model';
import { TeamMembersService } from './team-members.service';

@Component({
  selector: 'app-team-members',
  templateUrl: './team-members.component.html',
  styleUrls: ['./team-members.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TeamMembersComponent {
  readonly memberCards$: Observable<MemberCards> = this.teamMembersService.getTeamMembers();

  constructor(private readonly teamMembersService: TeamMembersService) {}
}
