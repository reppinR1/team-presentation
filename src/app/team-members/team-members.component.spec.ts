import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { TeamMembersComponent } from './team-members.component';
import { TeamMembersService } from './team-members.service';
import { MockComponent } from 'ng-mocks';
import { TeamMemberCardGridComponent } from './team-member-card-grid/team-member-card-grid.component';
import { of } from 'rxjs';

describe('TeamMembersComponent', () => {
  let spectator: Spectator<TeamMembersComponent>;
  const mocks = {
    teamMembers: {},
  };
  const createComponent = createComponentFactory({
    component: TeamMembersComponent,
    shallow: true,
    declarations: [MockComponent(TeamMemberCardGridComponent)],
    providers: [
      {
        provide: TeamMembersService,
        useValue: {
          getTeamMembers: jest.fn(() => of(mocks.teamMembers)),
        },
      },
    ],
  });
  const selectors = {
    cardGridComponent: () => spectator.query(TeamMemberCardGridComponent),
  };

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should render title and team member cards grid', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
  it('should set <app-team-members-grid> [memberCards] to memberCards$', () => {
    expect(selectors.cardGridComponent().memberCards).toEqual(
      mocks.teamMembers
    );
  });
});
