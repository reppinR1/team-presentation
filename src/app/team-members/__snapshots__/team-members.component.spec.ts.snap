// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMembersComponent should render title and team member cards grid 1`] = `
<app-team-members
  __ngContext__={[Function LRootView]}
  memberCards$={[Function Observable]}
  teamMembersService={[Function Object]}
>
  <h1
    class="title"
  >
    Our team
  </h1><app-team-member-card-grid />
</app-team-members>
`;
